tests/com/twitter/intellij/pants/util/PantsUtilTest.java

```diff
@@ -36,4 +41,57 @@ public void run() {
     // Make sure they are identical, meaning that no new JDK was created on the 2nd find.
     assertTrue(sdkA.get() == sdkB.get());
   }
+
+  public void testListTargetsExistingNonBuildFile() {
+    File projectDir = getProjectFolder();
+    VirtualFile virtualProjectDir = LocalFileSystem.getInstance().findFileByIoFile(projectDir);
+    Optional<VirtualFile> pantsIniFileResult = PantsUtil.findPantsIniFile(Optional.ofNullable(virtualProjectDir));
+    assertTrue(pantsIniFileResult.isPresent());
+    String pantsIniFilePath = pantsIniFileResult.get().getPath();
+    File pantsIniFile = new File(pantsIniFilePath);
+    assertTrue(pantsIniFile.exists() && !pantsIniFile.isDirectory());
+    assertTrue(!PantsUtil.isBUILDFilePath(pantsIniFilePath));
+    assertEquals(PantsUtil.listAllTargets(pantsIniFilePath),
+                 Lists.newArrayList());
+  }
+
+  public void testListTargetsNonexistentNonBuildFile() {
+    String nonexistentFilePath = "not/a/build/file/path";
+    File nonexistentFile = new File(nonexistentFilePath);
+    assertTrue(!nonexistentFile.exists());
+    assertTrue(!PantsUtil.isBUILDFilePath(nonexistentFilePath));
+    assertEquals(PantsUtil.listAllTargets(nonexistentFilePath),
+                 Lists.newArrayList());
+  }
+
+  public void testListTargetsNonexistentBuildFile() {
+    String nonexistentBuildFilePath = "not/a/real/BUILD";
+    File nonexistentBuildFile = new File(nonexistentBuildFilePath);
+    assertTrue(!nonexistentBuildFile.exists());
+    assertTrue(PantsUtil.isBUILDFilePath(nonexistentBuildFilePath));
+    boolean caught = false;
+    try {
+      PantsUtil.listAllTargets(nonexistentBuildFilePath);
+    } catch (PantsException e) {
+      caught = true;
+    }
+    assertTrue("PantsException not thrown for nonexistent BUILD file", caught);
+  }
+
+  public void testListTargetInvalidBuildFile() {
+    String projectDir = getProjectFolder().getPath();
+    Path invalidBuildFilePath = Paths.get(
+      projectDir, "../../testData/invalid-build-file/BUILD");
+    String invalidBuildFileLocation = invalidBuildFilePath.normalize().toString();
+    File invalidBuildFile = new File(invalidBuildFileLocation);
+    assertTrue(invalidBuildFile.exists() && !invalidBuildFile.isDirectory());
+    assertTrue(PantsUtil.isBUILDFilePath(invalidBuildFileLocation));
+    boolean caught = false;
+    try {
+      PantsUtil.listAllTargets(invalidBuildFileLocation);
+    } catch (PantsException e) {
+      caught = true;
+    }
+    assertTrue("PantsException not thrown for invalid BUILD file", caught);
```

Comment:
you can put the assertion/fail message in the try block, e.g. https://github.com/wisechengyi/intellij-pants-plugin/blob/e01fd6db7c00636108cb95f12bb8bef7fc853e59/tests/com/twitter/intellij/pants/model/PantsOptionsTest.java#L53-L61

so no need for the `caught` variable.

Reply:
Thanks for the example, I hadn't thought of that. I'll make the change.